source/common/log_sink/log_obj_store.h

```diff
@@ -0,0 +1,112 @@
+#pragma once
+
+#include <vector>
+
+#include "common/common/assert.h"
+
+//#include "envoy/access_log/access_log.h"
+
+namespace Envoy {
+namespace LogTransport {
+
+/**
+ * Provide a generic wrapper for a block of multiple log objects (of any type).
```

Comment:
That was the idea.  This would reduce contention on the queue by reducing insertion frequency and queue operations.  There could still be more batching if desired before shipping to the network.